### PR TITLE
Add conftest.py and created a tests folder

### LINKED ISSUE #66: Add conftest.py

### PR DESCRIPTION
**Pull request by idosinger98**
I created a file that holds all the common fixtures.
Resolve #66

### CODE REVIEW

**Review by patrisiakaplun — APPROVED** [review]
LGTM

**Review by Yarboa — COMMENTED** [review]
What about updating all tests to consume the fixtures? 
This will prove that fixture is used and tested, dont you think?

Other option could be open tickets for all the required changes in each app
In case you choose this one,  mentions all Issues opened to this PR

**Comment by idosinger98** [comment]
> What about updating all tests to consume the fixtures? This will prove that fixture is used and tested, dont you think?
> 
> Other option could be open tickets for all the required changes in each app In case you choose this one, mentions all Issues opened to this PR

I tested it on my computer according to the changes I made to reservation.test.py and it worked. So that everyone could test it, it needs to be merged, unless you want us to add more tests to this PR and then you can see that it works. I think the second option of opening an issues is better

**Comment by Yarboa** [comment]
> > What about updating all tests to consume the fixtures? This will prove that fixture is used and tested, dont you think?
> > Other option could be open tickets for all the required changes in each app In case you choose this one, mentions all Issues opened to this PR
> 
> I tested it on my computer according to the changes I made to reservation.test.py and it worked. So that everyone could test it, it needs to be merged, unless you want us to add more tests to this PR and then you can see that it works. I think the second option of opening an issues is better

I assume, planning in advance could collect the common fixtures exist in main branch, add then to existing 
like those

https://github.com/redhat-beyond/Pro4U/blob/main/account/tests.py#L29-L75

New components should use the Client/Professional  fixtures, don't you think?

**Comment by idosinger98** [comment]
> > > What about updating all tests to consume the fixtures? This will prove that fixture is used and tested, dont you think?
> > > Other option could be open tickets for all the required changes in each app In case you choose this one, mentions all Issues opened to this PR
> > 
> > 
> > I tested it on my computer according to the changes I made to reservation.test.py and it worked. So that everyone could test it, it needs to be merged, unless you want us to add more tests to this PR and then you can see that it works. I think the second option of opening an issues is better
> 
> I assume, planning in advance could collect the common fixtures exist in main branch, add then to existing like those
> 
> https://github.com/redhat-beyond/Pro4U/blob/main/account/tests.py#L29-L75
> 
> New components should use the Client/Professional fixtures, don't you think?
> 
> **Please note:** No tests, no option to Review, In case you have a goal, at least update/move existing tests to use this fixture

But i add Client/Professional fixtures.

**Comment by Yarboa** [comment]
> > > > What about updating all tests to consume the fixtures? This will prove that fixture is used and tested, dont you think?
> > > > Other option could be open tickets for all the required changes in each app In case you choose this one, mentions all Issues opened to this PR
> > > 
> > > 
> > > I tested it on my computer according to the changes I made to reservation.test.py and it worked. So that everyone could test it, it needs to be merged, unless you want us to add more tests to this PR and then you can see that it works. I think the second option of opening an issues is better
> > 
> > 
> > I assume, planning in advance could collect the common fixtures exist in main branch, add then to existing like those
> > https://github.com/redhat-beyond/Pro4U/blob/main/account/tests.py#L29-L75
> > New components should use the Client/Professional fixtures, don't you think?
> > **Please note:** No tests, no option to Review, In case you have a goal, at least update/move existing tests to use this fixture
> 
> But i add Client/Professional fixtures.

Is it running by any test? 
Did you remove the previous fixtures? which are local?

**Comment by idosinger98** [comment]
> > > > > What about updating all tests to consume the fixtures? This will prove that fixture is used and tested, dont you think?
> > > > > Other option could be open tickets for all the required changes in each app In case you choose this one, mentions all Issues opened to this PR
> > > > 
> > > > 
> > > > I tested it on my computer according to the changes I made to reservation.test.py and it worked. So that everyone could test it, it needs to be merged, unless you want us to add more tests to this PR and then you can see that it works. I think the second option of opening an issues is better
> > > 
> > > 
> > > I assume, planning in advance could collect the common fixtures exist in main branch, add then to existing like those
> > > https://github.com/redhat-beyond/Pro4U/blob/main/account/tests.py#L29-L75
> > > New components should use the Client/Professional fixtures, don't you think?
> > > **Please note:** No tests, no option to Review, In case you have a goal, at least update/move existing tests to use this fixture
> > 
> > 
> > But i add Client/Professional fixtures.
> 
> Is it running by any test? Did you remove the previous fixtures? which are local?

Yes I removed the previous fixtures from my local computer and ran the tests again and it worked. After this PR is merged, everyone will change their test.py according to the conftest.py file. Then you can see that the tests work. Now I add to the conftest.py file the common functions that appear in review/test.py

**Comment by Yarboa** [comment]
> > > > > > What about updating all tests to consume the fixtures? This will prove that fixture is used and tested, dont you think?
> > > > > > Other option could be open tickets for all the required changes in each app In case you choose this one, mentions all Issues opened to this PR
> > > > > 
> > > > > 
> > > > > I tested it on my computer according to the changes I made to reservation.test.py and it worked. So that everyone could test it, it needs to be merged, unless you want us to add more tests to this PR and then you can see that it works. I think the second option of opening an issues is better
> > > > 
Please, what ever you tested locally and useful for all apps, 
should be done here,
At least update account testing inorder to verify it is working,

> > > > 
> > > > I assume, planning in advance could collect the common fixtures exist in main branch, add then to existing like those
> > > > https://github.com/redhat-beyond/Pro4U/blob/main/account/tests.py#L29-L75
> > > > New components should use the Client/Professional fixtures, don't you think?
> > > > **Please note:** No tests, no option to Review, In case you have a goal, at least update/move existing tests to use this fixture
> > > 
> > > 
> > > But i add Client/Professional fixtures.
> > 
> > 
> > Is it running by any test? Did you remove the previous fixtures? which are local?
> 
> Yes I removed the previous fixtures from my local computer and ran the tests again and it worked. After this PR is merged, everyone will change their test.py according to the conftest.py file. Then you can see that the tests work. Now I add to the conftest.py file the common functions that appear in review/test.py

Sorry, tests are mandatory for this PR

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+
+@pytest.fixture
+def persisted_typeOfJob(typeOfJob):
+    typeOfJob.professional_id.save()
+    typeOfJob.save()
+    return typeOfJob
+
+
+@pytest.fixture
```
Comment (Yarboa) [inline tests/conftest.py:107]: Please try to focus on Profile/Client fixtures,
It is fundamental, other PRs will have to rebase with it.

File: account/tests.py [hunk 2]
```diff
@@ -1,105 +1,31 @@
...
-@pytest.fixture
-def client(profile):
-    client_profile = profile
-    client_profile.user_type = CLIENT_TYPE,
-    client = Client.objects.create(
-        profile_id=profile,
-        birthday=BIRTHDAY
-    )
-    return client
+from tests.conftest import *
```
Comment (Yarboa) [inline account/tests.py:1]: It should be import from upper hierarchy.
It is searching conftest in account

Comment (idosinger98) [reply account/tests.py]: Done

File: tests/conftest.py (hunk 1 above)
Comment (idosinger98) [reply tests/conftest.py:101]: Should there only be functions related to the account app?

Comment (Yarboa) [reply tests/conftest.py:101]: Yes, if no tests now, using the fixtures,. please remove

File: tests/account/tests.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
+from account.models.profile import Profile
```
Comment (Yarboa) [inline tests/account/tests.py:1]: The location should stay under account please do not move it

**Review by Yarboa — CHANGES_REQUESTED** [review]

File: account/tests.py (hunk 2 above)
Comment (Yarboa) [reply account/tests.py]: Please do not change file location

**Comment by guybeckenstein** [comment]
Is tests/conftest.py file within account app?

File: tests/account/tests.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+from account.models.professional import Professions, Professional
+from account.models.client import Client
+from datetime import datetime
+import pytest
+
+PROFESSIONAL_TYPE = 'C'
```
Comment (TalReinfeld) [inline tests/account/tests.py:7]: Just change it to PROFILE_TYPE

File: tests/account/tests.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+CITY = 'New York'
+ADDRESS = '123 Main St'
+DESCRIPTION = 'I'
+BIRTHDAY = datetime(2000, 1, 1)
+
+
+@pytest.mark.django_db
+class TestProfileModel:
+    def test_new_profile(self, profile1):
+        assert profile1.user_type == PROFESSIONAL_TYPE
```
Comment (TalReinfeld) [inline tests/account/tests.py:20]: Also here - PROFILE_TYPE

File: tests/account/tests.py (hunk 4 above)
Comment (idosinger98) [reply tests/account/tests.py]: Done

File: tests/account/tests.py (hunk 5 above)
Comment (idosinger98) [reply tests/account/tests.py]: Done

**Review by TalReinfeld — APPROVED** [review]
LGTM

File: account/tests.py (hunk 2 above)
Comment (Yarboa) [reply account/tests.py]: renamed:    tests/account/tests.py -> account/tests.py
	renamed:    tests/conftest.py -> conftest.py

account/tests.py
 import pytest
+import conftest

Tested it

File: tests/conftest.py (hunk 1 above)
Comment (Yarboa) [reply tests/conftest.py]: Please remove it

Comment (idosinger98) [reply tests/conftest.py]: Done

File: account/tests.py (hunk 2 above)
Comment (idosinger98) [reply account/tests.py]: Done

**Review by patrisiakaplun — APPROVED** [review]
LGTM

**Review by idoyek — COMMENTED** [review]
@idosinger98 Mabye squash\fixup commits to 1 or 2 before the PR merges, you have a lot

**Review by Yarboa — APPROVED** [review]
LGTM

**Pull request merged**